### DISCUSSION TITLE
error when compiled lammps with deepmd package

### DISCUSSION THREAD

**Discussion by zfan2016** [deepmodeling/deepmd-kit#220] [Q&A]
Hello,

I successfully built the python interfaces as your instructions.
And during building c++ interfaces, there is no any error. But when I checked $deepmd_root/lib

ls $deepmd_root/lib
libdeepmd.a libdeepmd_ipi.a libdeepmd_op.a

which is not consistent with your
 libdeepmd_ipi.so libdeepmd_op.so libdeepmd.so

But I can see the directory USER-DEEPMD. So I copied it to /src of lammps and then compiled lammps, but I got the error below:

$ make yes-user-deepmd
Installing package user-deepmd
$ make mpi
make[1]: Entering directory '/global/u2/z/zfan/software/lammps-31Mar17_deepmd/src/Obj_mpi'
make[1]: *** No rule to make target 'depend'.  Stop.
make[1]: Leaving directory '/global/u2/z/zfan/software/lammps-31Mar17_deepmd/src/Obj_mpi'
make[1]: Entering directory '/global/u2/z/zfan/software/lammps-31Mar17_deepmd/src/Obj_mpi'
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -M ../fix_dplr.cpp > fix_dplr.d
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -M ../force.cpp > force.d
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -M ../lammps.cpp > lammps.d
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -M ../modify.cpp > modify.d
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -M ../pair_nnp.cpp > pair_nnp.d
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -M ../pppm_dplr.cpp > pppm_dplr.d
CC -g -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd    -DLAMMPS_GZIP -std=c++11 -DHIGH_PREC   -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/tensorflow_root/include -I/global/cscratch1/sd/zfan/deepmd/deepmd_root/include/deepmd   -DFFT_NONE   -c ../pppm_dplr.cpp

../pppm_dplr.cpp(48): error: no instance of constructor "LAMMPS_NS::PPPM::PPPM" matches the argument list
            argument types are: (LAMMPS_NS::LAMMPS *)
    PPPM(lmp)
        ^
../pppm.h(38): note: this candidate was rejected because arguments do not match
  class PPPM : public KSpace {
        ^
../pppm.h(40): note: this candidate was rejected because mismatch in count of arguments
    PPPM(class LAMMPS *, int, char **);
    ^
compilation aborted for ../pppm_dplr.cpp (code 2)
make[1]: *** [Makefile:109: pppm_dplr.o] Error 2
make[1]: Leaving directory '/global/u2/z/zfan/software/lammps-31Mar17_deepmd/src/Obj_mpi'
make: *** [Makefile:144: mpi] Error 2

Could you please give me some suggestions to resolve the issue? Thanks.

**Comment by njzjz** [comment]
If you use code on devel branch, please enable the kspace package in LAMMPS.

**Comment by zfan2016** [comment]
But I still got the similar error after added "make yes-kspace".

**Comment by njzjz** [comment]
Maybe your LAMMPS is too old. Try to use a latest LAMMPS.
